nannyml/data_quality/range/calculator.py

```diff
@@ -0,0 +1,252 @@
+#  Author:   James Nesfield  <jamesnesfield@live.com>
+#
+#  License: Apache Software License 2.0
+
+"""Continous numerical variable range monitor to ensure range supplied is within training bounds."""
+
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+from pandas import MultiIndex
+
+from nannyml.base import AbstractCalculator, _list_missing, _split_features_by_type
+from nannyml.chunk import Chunker
+from nannyml.exceptions import InvalidArgumentsException
+from nannyml.thresholds import StandardDeviationThreshold, Threshold, calculate_threshold_values
+from nannyml.usage_logging import UsageEvent, log_usage
+
+from .result import Result
+
+"""
+Values Out Of Range Data Quality Module.
+"""
+
+
+class NumericalRangeCalculator(AbstractCalculator):
+    """NumericalRangeCalculator implementation to ensure inference data numerical ranges match training."""
+
+    def __init__(
+        self,
+        column_names: Union[str, List[str]],
+        normalize: bool = True,
+        timestamp_column_name: Optional[str] = None,
+        chunk_size: Optional[int] = None,
+        chunk_number: Optional[int] = None,
+        chunk_period: Optional[str] = None,
+        chunker: Optional[Chunker] = None,
+        threshold: Threshold = StandardDeviationThreshold(),
```

Comment:
I think it makes way more sense to go for a `ConstantThreshold` here. We basically need any non-zero value for the "out of range count / rate" to cause an alert. This would translate into a `ConstantThreshold` with an upper threshold of 0, and no lower threshold (theoretically...).